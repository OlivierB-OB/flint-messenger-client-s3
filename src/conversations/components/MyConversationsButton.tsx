import Badge from '@material-ui/core/Badge';
import IconButton from '@material-ui/core/IconButton';
import Forum from '@material-ui/icons/Forum';
import React from 'react';
import { connect } from 'react-redux';
import { Action } from 'redux';
import { ThunkDispatch } from 'redux-thunk';
import { IAppState } from '../../appReducer';
import { makeShowConversationList } from '../actions/makeShowConversationList';

interface IShowConversationsButtonProps {
  unseenMessages: number;
  showConversationList: () => void;
}

export function ShowConversationsButton({ unseenMessages, showConversationList }: IShowConversationsButtonProps) {
  return (
    <IconButton aria-label="contacts" onClick={showConversationList}>
      <Badge badgeContent={unseenMessages} color="error" anchorOrigin={{ vertical: 'top', horizontal: 'left' }}>
        <Forum fontSize="large" />
      </Badge>
    </IconButton>
  );
}

const mapStateToProps = ({ conversations }: IAppState) => ({
  unseenMessages: conversations.unseenMessages,
});

const mapDispatchToProps = (dispatch: ThunkDispatch<IAppState, void, Action>) => ({
  showConversationList: () => dispatch(makeShowConversationList()),
});

export const MyConversationsButton = connect(mapStateToProps, mapDispatchToProps)(ShowConversationsButton);
