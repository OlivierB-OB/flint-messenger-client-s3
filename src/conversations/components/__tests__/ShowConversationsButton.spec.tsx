import React from 'react';
import { shallow } from 'enzyme';
import { ShowConversationsButton } from '../MyConversationsButton';

describe('ShowConversationsButton', () => {
  it('should display a show conversation button...', async () => {
    const props: any = {
      unseenMessages: 'nb unseen messages',
      showConversationList: jest.fn(),
    }
    const component = shallow(<ShowConversationsButton {...props} />);
    expect(component).toMatchSnapshot();
    component.simulate('click');
    expect(props.showConversationList).toHaveBeenCalled();
  });
});
