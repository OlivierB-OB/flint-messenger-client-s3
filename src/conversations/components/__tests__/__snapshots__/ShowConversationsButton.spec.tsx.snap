// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowConversationsButton should display a show conversation button... 1`] = `
<WithStyles(ForwardRef(IconButton))
  aria-label="contacts"
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(Badge))
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    badgeContent="nb unseen messages"
    color="error"
  >
    <ForumIcon
      fontSize="large"
    />
  </WithStyles(ForwardRef(Badge))>
</WithStyles(ForwardRef(IconButton))>
`;
